### ISSUE TITLE
step1 参数更新时间超长

### ISSUE THREAD

**Issue by X-jun-0130** [volcengine/verl#496]
set -x
export VLLM_ATTENTION_BACKEND=XFORMERS

python3 /LLM-veRL/Model_GRPO/main_grpo.py \
    algorithm.adv_estimator=grpo \
    data.train_files=/LLM-veRL/Model_Train_Data/train_grpo.parquet \
    data.val_files=/LLM-veRL/Model_Train_Data/test_grpo.parquet \
    data.train_batch_size=64 \
    data.val_batch_size=64 \
    data.max_prompt_length=1024 \
    data.max_response_length=8192 \
    actor_rollout_ref.model.path=/workspace/Qwen2.5-32B-Instruct \
    actor_rollout_ref.actor.optim.lr=1e-6 \
    actor_rollout_ref.model.use_remove_padding=True \
    actor_rollout_ref.actor.ppo_mini_batch_size=16 \
    actor_rollout_ref.actor.ppo_micro_batch_size=2 \
    actor_rollout_ref.actor.ulysses_sequence_parallel_size=4 \
    actor_rollout_ref.actor.use_kl_loss=True \
    actor_rollout_ref.actor.kl_loss_coef=0.001 \
    actor_rollout_ref.actor.kl_loss_type=low_var_kl \
    actor_rollout_ref.model.enable_gradient_checkpointing=True \
    actor_rollout_ref.actor.fsdp_config.param_offload=True \
    actor_rollout_ref.actor.fsdp_config.grad_offload=True \
    actor_rollout_ref.actor.fsdp_config.optimizer_offload=True \
    actor_rollout_ref.rollout.temperature=0.6 \
    actor_rollout_ref.rollout.top_p=0.95 \
    actor_rollout_ref.rollout.log_prob_micro_batch_size=128 \
    actor_rollout_ref.rollout.tensor_model_parallel_size=4 \
    actor_rollout_ref.rollout.name=vllm \
    actor_rollout_ref.rollout.gpu_memory_utilization=0.8 \
    actor_rollout_ref.rollout.n=16 \
    actor_rollout_ref.ref.log_prob_micro_batch_size=128 \
    actor_rollout_ref.ref.fsdp_config.param_offload=True \
    algorithm.kl_ctrl.kl_coef=0.001 \
    trainer.critic_warmup=0 \
    trainer.logger=['console','tensorboard']  \
    trainer.project_name='verl_grpo_wingpt' \
    trainer.experiment_name='qwen25_32b' \
    trainer.n_gpus_per_node=8 \
    trainer.nnodes=1 \
    trainer.save_freq=72 \
    trainer.test_freq=72 \
    trainer.total_epochs=3 $@

step:1 - global_seqlen/min:74428.000 - global_seqlen/max:135452.000 - global_seqlen/minmax_diff:61024.000 - global_seqlen/balanced_min:101038.000 - global_seqlen/balanced_max:101039.000 - global_seqlen/mean:101038.125 - actor/kl_loss:0.001 - actor/kl_coef:0.001 - actor/entropy_loss:0.155 - actor/pg_loss:-0.001 - actor/pg_clipfrac:0.003 - actor/ppo_kl:-0.000 - actor/grad_norm:0.605 - mfu/actor:0.008 - actor/lr:0.000 - critic/score/mean:-0.151 - critic/score/max:1.000 - critic/score/min:-1.000 - critic/rewards/mean:-0.151 - critic/rewards/max:1.000 - critic/rewards/min:-1.000 - critic/advantages/mean:-0.320 - critic/advantages/max:3.750 - critic/advantages/min:-3.750 - critic/returns/mean:-0.320 - critic/returns/max:3.750 - critic/returns/min:-3.750 - response_length/mean:595.517 - response_length/max:8192.000 - response_length/min:69.000 - response_length/clip_ratio:0.013 - prompt_length/mean:193.844 - prompt_length/max:891.000 - prompt_length/min:76.000 - prompt_length/clip_ratio:0.000 - timing_s/gen:383.238 - timing_s/old_log_prob:52.481 - timing_s/ref:75.377 - timing_s/adv:2.163 - timing_s/update_actor:8659.013 - timing_s/step:9172.412 - timing_per_token_ms/gen:0.628 - timing_per_token_ms/ref:0.093 - timing_per_token_ms/adv:0.003 - timing_per_token_ms/update_actor:10.713

timing_s/update_actor:8659.013 参数更新时间竟然用了这么长的时间，这种情况正常吗？   机器配置8*A800 80G

**Comment by Yangruipis** [comment]
same problem

**Comment by 1074676247** [comment]
same problem

**Comment by xiusic** [comment]
Same problem

**Comment by wjn1996** [comment]
The same problem. 
When I choose 7B model with default settings (e.g., sp=4, max_len=1024*20, vllm + fsdp, rollout.n=16), the time cost per step is 45min and the time cost of update_actor is about 20min. When I reset rollout.n=4, the time cost of update_actor is 10min. What's the problem? 
@xiusic @Yangruipis @X-jun-0130 @1074676247 @pcmoritz

**Comment by Osilly** [comment]
The same problem. So how to solve it?

**Comment by AnchoretY** [comment]
同样的问题，有人知道怎么解决吗

**Comment by gray311** [comment]
same problem

**Comment by wxf12345** [comment]
same problem

**Comment by ljy2222** [comment]
same problem

**Comment by PorkFormer** [comment]
same problem

**Comment by Monohydroxides** [comment]
same problem

**Comment by caixd-220529** [comment]
same proble

**Comment by Kirrito-k423** [comment]
中文：

从当前公开信息看，`timing_s/update_actor` 不是单纯“参数更新时间”，它覆盖了 actor 侧 PPO mini-batch 训练段。代码里 `_update_actor` 会把 `actor_rollout_ref.actor.ppo_mini_batch_size` 乘以 `actor_rollout_ref.rollout.n` 后传给 worker；worker 的 `update_actor` 最终执行 `actor.train_mini_batch(...)`。因此在这个配置里，`ppo_mini_batch_size=16` 且 `rollout.n=16`，actor 更新阶段实际会处理放大的响应样本规模；再叠加 32B 模型、`max_response_length=8192`、gradient checkpointing，以及 actor 的 param/grad/optimizer offload，都可能让 `update_actor` 成为主要瓶颈。

为了判断这是配置导致的预期慢速、硬件/通信瓶颈，还是 verl 的性能问题，麻烦补充：
1. verl commit/version、PyTorch/CUDA/vLLM 版本；
2. `update_actor` 内部 profiling 或至少 forward/backward/optimizer/offload/communication 的分项耗时；
3. 对比实验：将 `rollout.n` 降到 4 或 8、关闭 actor optimizer/grad/param offload（如果显存允许）、调整 `ppo_micro_batch_size_per_gpu` 后的 step/update_actor 时间；
4. GPU 利用率、CPU 内存/PCIe 或 NVLink 传输情况。

English:

Based on the public evidence, `timing_s/update_actor` is not only the parameter-update instant. It wraps the actor-side PPO mini-batch training path. In the trainer, `actor.ppo_mini_batch_size` is multiplied by `rollout.n`, and the worker-side `update_actor` calls `actor.train_mini_batch(...)`. With `ppo_mini_batch_size=16`, `rollout.n=16`, a 32B model, `max_response_length=8192`, gradient checkpointing, and actor param/grad/optimizer offload, it is plausible for `update_actor` to dominate the step time.

To distinguish expected configuration overhead from a verl performance issue, please provide:
1. verl commit/version plus PyTorch/CUDA/vLLM versions;
2. profiling inside `update_actor`, or at least forward/backward/optimizer/offload/communication breakdown;
3. comparison runs with `rollout.n=4/8`, actor offload disabled if memory allows, and adjusted `ppo_micro_batch_size_per_gpu`;
4. GPU utilization and CPU memory/PCIe/NVLink transfer observations.

This response was generated by a scheduled Codex GPT5.5 issue processing task.
